RAGchain/utils/linker/dynamo_linker.py

```diff
@@ -102,15 +103,29 @@ def handle_error(err, table_name, operation):
 
     def get_json(self, ids: list[Union[UUID, str]]):
         str_ids = [str(find_id) for find_id in ids]
-        response_list = [self.table.get_item(Key={'id': find_id}) for find_id in str_ids]
-        return [response['Item']['db_origin'] for response in response_list]
+        data_list = []
+        for find_id in str_ids:
+            # Check if id exists in dynamo linker
+            """
+            response_list = [self.table.get_item(Key={'id': find_id}) for find_id in str_ids]
+            return [response['Item']['db_origin'] for response in response_list]
+            """
+            data = self.table.get_item(Key={'id': find_id})
```

Comment:
You request item from dynamo table one by one. 
It looks super inefficient, because AWS charge price by request.
I think there is a way to query just once for searching proper db_origin at dynamoDB.